### PR TITLE
Pull 64-bit windows support changes into omnibus-chef

### PR DESCRIPTION
**Pull request by ksubrama**

### CODE REVIEW

**Comment by ksubrama** [comment]
This supercedes #451

File: omnibus.rb [hunk 1]
```diff
@@ -24,6 +24,10 @@
 # package_dir   './local/omnibus/pkg'
 # package_tmp   './local/omnibus/pkg-tmp'
 
+# Windows architecture defaults - set to x86 unless otherwise specified.
+# ------------------------------
+windows_arch   ENV['OMNIBUS_WINDOWS_ARCH'] || :x86
```
Comment (danielsdeleo) [inline omnibus.rb]: Curious why we're setting an environment variable here. Just so we can do one-off builds locally w/ 64 bit? Also, environment variables will always be strings, but the default is a symbol. Probably won't matter but would be more consistent for the default to be "x86".

Comment (ksubrama) [inline omnibus.rb]: I think we've been using :x86 throughout omnibus software.  I think I'll do a :to_sym and a quick check instead.  The environment variable is to assist eng services.  They can bake the 64-bit windows builders with this environment variable instead of having to patch this file every time they clone.

File: Gemfile [hunk 2]
```diff
@@ -12,7 +12,7 @@ group :development do
   gem 'berkshelf', '~> 3.0'
 
   # Use Test Kitchen with Vagrant for converging the build environment
-  gem 'test-kitchen',    '~> 1.4.0.rc.1'
-  gem 'kitchen-vagrant', '~> 0.17.0.rc.1'
+  gem 'test-kitchen',    '~> 1.4.0'
```
Comment (jaym) [inline Gemfile]: Unrelated change. Perhaps do this in another PR. You should test that the things still work when doing this

File: config/projects/chef.rb [hunk 3]
```diff
@@ -40,20 +40,14 @@
...
-######
+
+override :'ruby-windows', version: "2.0.0-p645"
 override :rubygems,       version: "2.4.4"
 
 # Chef Release version pinning
 #override :chef, version: "12.4.1"
 #override :ohai, version: "8.5.0"
-override :'openssl-windows', version: '1.0.0r'
+
```
Comment (jaym) [inline config/projects/chef.rb]: why remove this?

File: Gemfile (hunk 2 above)
Comment (ksubrama) [inline Gemfile]: It does - the existing Gemfile.lock already pulls in a more recent test kitchen.  Since I needed to run bundle update to pull in updated deps for omnibus as is, I figured I'd just clean up the "rc" stuff because I found it ugly :)

File: config/projects/chef.rb (hunk 3 above)
Comment (ksubrama) [inline config/projects/chef.rb]: Because that's the default openssl-windows version now.  I can leave it as that explicitly if you'd like.  Both chef and chef-dk should be using that version.

Comment (jaym) [inline config/projects/chef.rb]: :+1: if it's the default

File: Gemfile (hunk 2 above)
Comment (jaym) [inline Gemfile]: ahh, ok :+1:

**Comment by jaym** [comment]
:+1:
